### DISCUSSION TITLE
listaccounts is non-deterministic, despite the docs indicating it likely should be

### DISCUSSION THREAD

**Discussion by OliverNChalk** [blockscout/blockscout#5960] [Support] [answered]
I am hitting the following endpoint `?module=account&action=listaccounts` and when I query 10 pages with 10 records per page, I get non-deterministic results. The docs claim that this endpoint should return results sorted by when they were indexed by the block explorer.

Given the results are sorted, I would have imagined that this endpoint would be deterministic? e.g. the same params (page=x, offset=y) always yield the same result. Before I dig deeper and open an issue, would a non-deterministic result be considered a bug here?

**Comment by vbaranov** [comment, marked as answer]
Thank you for rising this question. Indeed, addresses are sorted by the time of insertion https://github.com/blockscout/blockscout/blob/e303d72f675a6fc8327ada509984025708026585/apps/explorer/lib/explorer/etherscan/addresses.ex#L21. And indeed, on different runs of mentioned API endpoint the order of results is changing. And this non-determinism happens because some addresses has exactly the insertion timestamp:
```
select inserted_at from addresses where hash in ('\x1a9c379a297134135D89A08CA9427f6DfB1a081D', '\xF16AE3939552e336852F55FE7a8402d4eE5c9a8c');
        inserted_at
---------------------------
 2018-10-06 22:36:19.65109
 2018-10-06 22:36:19.65109
(2 rows)
```

This happens most likely because multiple addresses during indexing process are inserted in a single DB transaction. That prompts the thought that we need one more sorting parameter for this API endpoint. I will create a bug for that.

Update: issue https://github.com/blockscout/blockscout/issues/6023 created.
